javascript/extractor/src/com/semmle/js/dependencies/VersionConstraint.java

```diff
@@ -0,0 +1,57 @@
+package com.semmle.js.dependencies;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VersionConstraint {
+    private String operator;
+    private SemVer version;
+
+    public VersionConstraint(String operator, SemVer version) {
+        this.operator = operator;
+        this.version = version;
+    }
+
+    public static VersionConstraint exact(String version) {
+        return new VersionConstraint("", SemVer.tryParse(version));
+    }
+
+    public String getOperator() {
+        return operator;
+    }
+
+    public SemVer getVersion() {
+        return version;
+    }
+
+    private static final Pattern pattern = Pattern.compile("([~^<>=]*)\\s*(\\d.*)");
+
+    public static List<VersionConstraint> parseVersionConstraints(String str) {
+        String[] parts = str.split(",");
```

Comment:
I don't get this line. 
[The documentation](https://docs.npmjs.com/files/package.json#dependencies) suggests that `||` and not `,` is used for multiple constraints.

Reply:
Oh you're right that's completely wrong 🤦